tests/test_web_poet_rules.py

```diff
@@ -1489,6 +1493,59 @@ def test_page_object_with_item_dependency_deadlock_2_d(caplog) -> None:
     assert "ProviderDependencyDeadlockError" in caplog.text
 
 
+@attrs.define
+class Mobius:
+    name: str
+
+
+@handle_urls(URL)
+@attrs.define
+class MobiusPage(WebPage[Mobius]):
+    mobius_item: Mobius
+
+    @field
+    def name(self) -> str:
+        return f"(modified) {self.mobius_item.name}"
+
+
+class MobiusProvider(PageObjectInputProvider):
+    provided_classes = {Mobius}
+
+    def __call__(self, to_provide: Set[Callable], request: scrapy.Request):
+        return [Mobius(name="mobius from MobiusProvider")]
+
+
+@inlineCallbacks
+def test_page_object_returning_item_which_is_also_a_dep() -> None:
+    settings = {
+        "SCRAPY_POET_PROVIDERS": {**DEFAULT_PROVIDERS, **{MobiusProvider: 1000}}
+    }
+
+    # item from 'to_return'
+    item, deps = yield crawl_item_and_deps(Mobius, override_settings=settings)
+    assert item == Mobius(name="(modified) mobius from MobiusProvider")
+    assert_deps(deps, {"item": Mobius})
+
+    # calling the actual page objects should still work
+    # item, deps = yield crawl_item_and_deps(MobiusPage)
+    # assert item == Mobius(name="(modified) mobius from MobiusProvider")
+    # assert_deps(deps, {"item": Mobius})
+
+
+# TODO: USE CASE: ProductItem as a dependency of ProductPage. ProductPage
+# returns ProductItem after modfiying some contents to it. ProductItem is
+# provided by:
+# - some provider
+# - some page objects in their ApplyRule
+
+# TODO: Same as the previous tests but multiple ApplyRules are available and top one is selected
+
+# TODO: Same as the 1st test but the settings is not overridden
+
+# TODO: Having the 2 new test cases above, update ``test_created_apply_rules``
+# below.
```

Comment:
TODO: address these remaining test scenarios.

Reply:
Done in https://github.com/scrapinghub/scrapy-poet/pull/164/commits/d148db2ddc0ad311b795dea810f4230e901134d3